planner/core/logical_plans.go

```diff
@@ -1582,6 +1587,8 @@ func (ds *DataSource) deriveIndexPathStats(path *util.AccessPath, _ []expression
 		} else {
 			path.CountAfterIndex = math.Max(path.CountAfterAccess*selectivity, ds.stats.RowCount)
 		}
+	} else {
+		path.CountAfterIndex = path.CountAfterAccess
```

Comment:
We don't set `path.CountAfterIndex` when `path.IndexFilters == nil` before the PR. Would that cause some potential wrong cardinality estimation? (since I thought `CountAfterIndex` would always be properly set before...

Reply:
That also surprised me...
It looks like there's no big problem most times, but I'm not sure about some edge cases, like when there are no index conditions here, but there are index conditions caused by virtual columns in `addPushedDownSelection`.